pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -719,6 +719,14 @@ cpdef inline str parse_timedelta_unit(str unit):
         return "ns"
     elif unit == "M":
         return unit
+    elif unit in {"MIN", "MS", "US", "NS"}:
+        warnings.warn(
+            f"\'{unit}\' is deprecated and will be removed in a future version. "
+            f"Please use \'{unit.lower()}\' instead of \'{unit}\'.",
+            FutureWarning,
+            stacklevel=find_stack_level(),
+        )
+        unit = unit.lower()
     elif unit in c_DEPR_ABBREVS:
```

Comment:
Instead of adding another branch is there any reason to not just add these to `c_DEPR_ABBREVS`?

Reply:
thank you, I agree, better to add these deprecated units to the dictionary. I made the changes.

I wonder, can we get these changes into `2.2.0`? The reason: we deprecated already uppercase strings `“MIN”, “MS”, “US”, and “NS” `denoting frequencies for `Period` and `offsets `in favor of lowercase strings.

Reply:
> we want to deprecate "MS" for "Milli" to avoid confusion.

sorry I might have missed something but where was this discussed? I don't remember it now. "month begin" isn't valid for timedelta, so I think there's no risk of confusion here?

> I wonder, can we get these changes into 2.2.x?

there's already 2.2.0 and 2.2.1 out, I think it's too late now

Reply:
As discussed: by

> deprecate "MS" for "Milli"

you meant "deprecate "MS" for milliseconds", not

> deprecate "MS" in favour of "Milli"

as I'd interpreted it

All good then 👍

Reply:
yeah, my wording wasn't clear enough. I meant: `deprecate "MS" for milliseconds"`.

I updated `Timedelta` docstring, and added a note to `v3.0.0`, but I am unsure if we need the note, because using uppercase strings `“MIN”, “MS”, “US”, “NS”` for units are not documented.

Reply:
thanks, sounds reasonable, I marked this PR as draft.